TF_Settings_Web/src/Components/Pages/Camera/Masking/MaskingScreen.tsx

```diff
@@ -162,62 +143,99 @@ const MaskingScreen = () => {
             socket.send(JSON.stringify({ type: 'SubscribeImageStreaming' }));
         }
     };
 
     const handleTFInput = (evt: CustomEvent<HandFrame>): void => {
-        if (isHandProcessingRef.current || !successfullySubscribed.current) {
-            return;
-        }
+        if (isHandProcessing.current || !successfullySubscribed.current) return;
 
-        isHandProcessingRef.current = true;
+        isHandProcessing.current = true;
 
         const hands = evt.detail?.Hands;
-        if (hands) {
+        if (hands.length > 0 || handData.current.one || handData.current.two) {
             const handOne = hands[0];
             const handTwo = hands[1];
             const convertedHandOne = handOne ? handToSvgData(handOne, 0) : undefined;
             const convertedHandTwo = handTwo ? handToSvgData(handTwo, 1) : undefined;
 
-            setHandData({ one: convertedHandOne, two: convertedHandTwo });
+            handData.current = { one: convertedHandOne, two: convertedHandTwo };
         }
 
         // Inore any messages for a short period to allow clearing of message handling
-        setTimeout(() => {
-            isHandProcessingRef.current = false;
+        handTimeoutRef.current = window.setTimeout(() => {
+            isHandProcessing.current = false;
         }, FRAME_PROCESSING_TIMEOUT);
     };
 
-    const timeoutFrame = () => {
-        frameTimeoutRef.current = window.setTimeout(() => {
-            setIsFrameProcessing(false);
-        }, FRAME_PROCESSING_TIMEOUT);
+    // ===== Components =====
+    const sliders = useMemo(
+        () =>
+            Object.entries(masking.current).map((sliderInfo) => (
+                <MaskingSliderDraggable
+                    key={sliderInfo[0]}
+                    direction={sliderInfo[0] as SliderDirection}
+                    maskingValue={sliderInfo[1]}
+                    canvasInfo={{ size: 800, topOffset: 50, leftOffset: 100 }}
+                    clearMasking={clearMasking}
+                    onDrag={setMasking}
+                    onDragEnd={sendMaskingRequest}
+                />
+            )),
+        [masking.current]
+    );
+
+    const lensToggles = useMemo(
+        () => (
+            <>
+                <MaskingLensToggle
+                    lens={'Left'}
+                    isMainLens={mainLens.current === 'Left'}
+                    setMainLens={(value: Lens) => (mainLens.current = value)}
+                />
+                <MaskingLensToggle
+                    lens={'Right'}
+                    isMainLens={mainLens.current === 'Right'}
+                    setMainLens={(value: Lens) => (mainLens.current = value)}
+                />
+            </>
+        ),
+        [mainLens.current]
+    );
+
+    const createMaskingOption = (input: MaskingOptionProps) => {
+        return (
+            <MaskingOption
+                title={input.title}
+                description={input.description}
```

Comment:
Can we remove this in favour of returning the JSX straight from the memos?

Reply:
Using straight JSX now (with memo)